controllers/chat/moveToAnon.js

```diff
@@ -54,102 +65,93 @@ const moveToAnon = async (req, res) => {
-            let new_anon_user_info_emoji_code =
-              checkChatAnonUserInfo?.anon_user_info_emoji_code || anon_user_info_emoji_code;
-            let new_anon_user_info_emoji_name =
-              checkChatAnonUserInfo?.anon_user_info_emoji_name || anon_user_info_emoji_name;
-
-            newStateMemberWithAnonInfo[member].anon_user_info_color_code =
-              new_anon_user_info_color_code;
-            newStateMemberWithAnonInfo[member].anon_user_info_color_name =
-              new_anon_user_info_color_name;
-            newStateMemberWithAnonInfo[member].anon_user_info_emoji_code =
-              new_anon_user_info_emoji_code;
-            newStateMemberWithAnonInfo[member].anon_user_info_emoji_name =
-              new_anon_user_info_emoji_name;
-
-            newStateMemberWithAnonInfo[member].user.name =
-              'Anonymous ' + new_anon_user_info_emoji_name;
-            newStateMemberWithAnonInfo[member].user.username =
-              'Anonymous ' + new_anon_user_info_emoji_name;
-
-            /**
-             * If the flag is true, insert new anon user info to DB
-             */
-            if (shouldNewAnonUserInfoInserted)
-              await ChatAnonUserInfo.create({
-                channel_id: newChannel.id,
-                my_anon_user_id: member,
-                target_user_id: targetUserId,
-                anon_user_info_color_code: new_anon_user_info_color_code,
-                anon_user_info_color_name: new_anon_user_info_color_name,
-                anon_user_info_emoji_code: new_anon_user_info_emoji_code,
-                anon_user_info_emoji_name: new_anon_user_info_emoji_name
-              });
-          }
-        }
-      })
-    );
+    //set target user detail
+    newStateMemberWithAnonInfo[prevTargetUser.user_id].user = prevTargetUser.user;
+    if (detailPrevTargetUser.is_anonymous) {
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_color_code =
+        prevTargetUser?.anon_user_info_color_code;
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_color_name =
+        prevTargetUser?.anon_user_info_color_name;
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_emoji_code =
+        prevTargetUser?.anon_user_info_emoji_code;
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_emoji_name =
+        prevTargetUser?.anon_user_info_emoji_name;
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: req.userId,
+        target_user_id: prevTargetUser.user_id,
+        anon_user_info_color_code: prevTargetUser?.anon_user_info_color_code,
+        anon_user_info_color_name: prevTargetUser?.anon_user_info_color_name,
+        anon_user_info_emoji_code: prevTargetUser?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: prevTargetUser?.anon_user_info_emoji_name
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code is setting the target user's details and creating a new `ChatAnonUserInfo` if the user is anonymous. However, it seems like there is no error handling in case any of these operations fail. It would be good to add some error handling here.
```diff
+ try {
     //set target user detail
     newStateMemberWithAnonInfo[prevTargetUser.user_id].user = prevTargetUser.user;
     if (detailPrevTargetUser.is_anonymous) {
       newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_color_code =
         prevTargetUser?.anon_user_info_color_code;
       newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_color_name =
         prevTargetUser?.anon_user_info_color_name;
       newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_emoji_code =
         prevTargetUser?.anon_user_info_emoji_code;
       newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_emoji_name =
         prevTargetUser?.anon_user_info_emoji_name;

       await ChatAnonUserInfo.create({
         channel_id: newChannel.id,
         my_anon_user_id: req.userId,
         target_user_id: prevTargetUser.user_id,
         anon_user_info_color_code: prevTargetUser?.anon_user_info_color_code,
         anon_user_info_color_name: prevTargetUser?.anon_user_info_color_name,
         anon_user_info_emoji_code: prevTargetUser?.anon_user_info_emoji_code,
         anon_user_info_emoji_name: prevTargetUser?.anon_user_info_emoji_name
       });
     }
+ } catch (error) {
+   console.error('Error setting target user details:', error);
+   throw error;
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->